src/Core/Context/CurrentContext.cs

```diff
@@ -137,6 +129,24 @@ public virtual Task SetContextAsync(ClaimsPrincipal user)
 
         var claimsDict = user.Claims.GroupBy(c => c.Type).ToDictionary(c => c.Key, c => c.Select(v => v));
 
+        ClientId = GetClaimValue(claimsDict, "client_id");
+
+        var clientType = GetClaimValue(claimsDict, Claims.Type);
+        if (clientType != null)
+        {
+            if (Enum.TryParse(clientType, out IdentityClientType c))
+            {
+                IdentityClientType = c;
+            }
+        }
+
+        if (IdentityClientType == IdentityClientType.Send)
+        {
+            // For the Send client, we don't need to set any User specific properties on the context
+            // so just short circuit and return here.
+            return Task.FromResult(0);
+        }
+
         var subject = GetClaimValue(claimsDict, "sub");
```

Comment:
The send access token technically has a subject - the send id.... does it make sense to move this above the return as well? We expect consumers to reference the send_id claim instead of the subject claim anyway.

Reply:
Oh yeah, good call.